dask/array/routines.py

```diff
@@ -256,6 +266,90 @@ def rot90(m, k=1, axes=(0, 1)):
         return flip(transpose(m, axes_list), axes[1])
 
 
+def _tensordot_rechunk(
+    lhs,
+    rhs,
+    left_axes,
+    right_axes,
+    out_shape,
+    out_index,
+    left_index,
+    right_index,
+    dtype=None,
+    limit=None,
```

Comment:
This function signature is a bit "everything and the kitchen sink". Everything does get used, but perhaps this could get tidied or split up?